sdk/tables/azure-data-tables/tests/test_challenge_auth.py

```diff
@@ -0,0 +1,321 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+from unittest.mock import Mock
+
+from azure.core.credentials import AccessToken
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.transport import HttpRequest as PipelineTransportHttpRequest
+from azure.core.rest import HttpRequest as RestHttpRequest
+from azure.data.tables._authentication import BearerTokenChallengePolicy
+import pytest
+
+HTTP_REQUESTS = [PipelineTransportHttpRequest, RestHttpRequest]
+
+
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
```

Comment:
Is it possible to maybe add a couple of 'playback only' recorded tests?

Reply:
Added!